app/policies/division_policy.rb

```diff
@@ -33,26 +33,27 @@ def select?
 
   class Scope < Scope
     def resolve
-      user ? scope.where(id: accessible_ids) : scope.published
+      if user
+        resolve_admin
+      else
+        resolve_public
+      end
     end
 
-    # This merges in child divisions of the divisions for which a user has been specifically
-    # granted access.
-    def accessible_ids
-      all_ids = base_accessible_ids.map do |id|
-        division = Division.find_safe(id)
-        division.self_and_descendants.pluck(:id) if division
-      end
-      all_ids.flatten.uniq.compact
+    def resolve_admin
+      scope.where(id: user.accessible_division_ids)
     end
 
-    # List of division hierarchy nodes for which user has been granted access.
-    def base_accessible_ids
-      user.roles.where(resource_type: :Division, name: [:member, :admin]).pluck(:resource_id).uniq
+    def resolve_public
+      scope.published
     end
 
     def accessible_divisions(public_only: false)
-      public_only ? Division.published : user.accessible_divisions
+      if public_only
+        resolve_public
+      else
+        resolve_admin
+      end
```

Comment:
Nice refactors here. I think you could perhaps go a bit further. The similarity of this method to `resolve` got me thinking. I think `public_only` is only ever true in the public loans case. The rest of the usages of this method via `division_select_options` are in the admin module. So I think really this method is just equivalent to `resolve` and it could be removed. Feel free to break my logic!